File: CST8319-DoctorAppointmentBookingSystem/src/java/model/Doctor.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package model;

/**
 * JavaBean for Doctor object.
 * @author fwu
 */
public class Doctor extends User {

    public Doctor(int id, String name, String address, String mobile, String email, String password) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.mobile = mobile;
        this.email = email;
        this.password = password;
    }
}

File: CST8319-DoctorAppointmentBookingSystem/web/JS/doctor.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */
document.addEventListener('DOMContentLoaded', () => {
    const noteButtons = document.querySelectorAll('.note-button');
    const noteModal = document.getElementById('noteModal');
    const closeModal = document.querySelector('.modal .close');
    const noteForm = document.getElementById('noteForm');
    const appointmentIDField = document.getElementById('appointmentID');

    noteButtons.forEach(button => {
        button.addEventListener('click', () => {
            const appointmentID = button.getAttribute('data-id');
            appointmentIDField.value = appointmentID;
            noteModal.style.display = 'block';
        });
    });

    closeModal.addEventListener('click', () => {
        noteModal.style.display = 'none';
    });

    window.onclick = function(event) {
        if (event.target === noteModal) {
            noteModal.style.display = 'none';
        }
    };
});


